# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1297033177 -3600
#      Sun Feb 06 23:59:37 2011 +0100
# Node ID 735e43b75e456b54ddfbce9a7714a782fa950746
# Parent  c60eaf7dac31b6f53305ae115173e39e63412bf8
Allow the legend location and orientation to be set without modifying the legend keys (#32374)

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,12 @@
 2011-02-05  David Bateman  <dbateman@free.fr>
 
+	* plot/legend.m: Allow the location and orientation to be set without
+	modifiying the legend keys.
 	* plot/legend.m: Add padding to legend positions. Fix for outerposition
 	for southeastoutside and southwestoutside.
 
 2011-02-06  Ben Abbott  <bpabbott@mac.com>
 
 	* plot/legend.m: Align legends to plot box, add demo. Bug 32373.
 	* plot/legend.m: "legend off" should delete the legend object.
 
diff --git a/scripts/plot/legend.m b/scripts/plot/legend.m
--- a/scripts/plot/legend.m
+++ b/scripts/plot/legend.m
@@ -283,16 +283,41 @@ function [hlegend2, hobjects2, hplot2, t
   elseif (strcmp (box, "on"))
     if (! isempty (hlegend))
       set (hlegend, "visible", "on", "box", "on");
     endif
   elseif (strcmp (box, "off"))
     if (! isempty (hlegend))
       set (hlegend, "box", "off", "visible", "off");
     endif
+  elseif (nargs == 0 && !(strcmp (position, "default") && 
+                          strcmp (orientation, "default")))
+    if (! isempty (hlegend))
+      hax = getfield (get (hlegend, "userdata"), "handle");
+      [hplots, text_strings] = __getlegenddata__ (hlegend);
+
+      if  (strcmp (position, "default"))
+        h = legend (hax, hplots, text_strings, "orientation", orientation);
+      elseif (strcmp (orientation, "default"))
+        if (outside)
+          h = legend (hax, hplots, text_strings, "location", 
+                      strcat (position, "outside"));
+        else
+          h = legend (hax, hplots, text_strings, "location", position);
+        endif
+      else
+        if (outside)
+          h = legend (hax, hplots, text_strings, "location", 
+                      strcat (position, "outside"), "orientation", orientation);
+        else
+          h = legend (hax, hplots, text_strings, "location", position,
+                      "orientation", orientation);
+        endif
+      endif
+    endif
   else
     hobjects = [];
     hplots  = [];
     text_strings = {};
 
     if (nargs > 0)
       have_data = false;
       for k = 1:nkids
